packages/datatrak-web/src/Routes.tsx

```diff
@@ -48,9 +63,10 @@ const LoggedInRedirect = ({ children }) => {
 
 // Reusable wrapper to handle redirecting to login if user is not logged in and the route is private
 const PrivateRoute = ({ children }: { children?: ReactNode }): any => {
-  const { isLoggedIn, isLoading, isFetched } = useUser();
+  const { isLoggedIn, isLoading, isFetched, data } = useUser();
   if (isLoading || !isFetched) return <FullPageLoader />;
   if (!isLoggedIn) return <Navigate to="/login" replace={true} />;
+  if (!data?.projectId) return <Navigate to={ROUTES.PROJECT_SELECT} replace={true} />;
```

Comment:
```suggestion
if (location.pathname === ROUTES.PROJECT_SELECT && data?.projectId) 
    return <Navigate to={ROUTES.HOME} replace={true} />;
  
  if (!data?.projectId) return <Navigate to={ROUTES.PROJECT_SELECT} replace={true} />;
```
The ProjectSelectRedirect seems so similar to the PrivateRoute and so it would be nice if we could just extend the private route component. This would clean up things a bit and also makes more sense to me since the ProjectSelectRedirect is still doing the Private Route check. It is a bit messy to add project id logic to the PrivateRoute but since we are already doing that, we might as well just do it agian.